kratos/includes/shared_pointers.h

```diff
@@ -68,18 +68,27 @@ std::ostream& operator <<(std::ostream& rOStream, const Kratos::intrusive_ptr<T>
 } // namespace Kratos
 
 
-#define KRATOS_CLASS_POINTER_DEFINITION(a) typedef Kratos::shared_ptr<a > Pointer; \
-typedef Kratos::shared_ptr<a > SharedPointer; \
-typedef Kratos::weak_ptr<a > WeakPointer; \
-typedef Kratos::unique_ptr<a > UniquePointer
+#define KRATOS_CLASS_POINTER_DEFINITION(a)                  \
+typedef Kratos::shared_ptr<a > Pointer;                     \
+typedef Kratos::shared_ptr<const a > PointerToConst;        \
+typedef Kratos::shared_ptr<a > SharedPointer;               \
+typedef Kratos::shared_ptr<const a > SharedPointerToConst;  \
+typedef Kratos::weak_ptr<a > WeakPointer;                   \
+typedef Kratos::weak_ptr<const a > WeakPointerToConst;      \
+typedef Kratos::unique_ptr<a > UniquePointer;               \
+typedef Kratos::unique_ptr<const a > UniquePointerToConst;
 
 namespace Kratos {
 template< class T > class GlobalPointer;
 }
 
-#define KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(a) typedef typename Kratos::intrusive_ptr<a > Pointer; \
-typedef Kratos::GlobalPointer<a > WeakPointer; \
```

Comment:
note for myself. I think this definition is wrong. To be checked in a different PR/issue